### PR TITLE
Relayer ContractReader( chain reader) constructor

### PR DESCRIPTION
**Pull request by ilija42**
https://github.com/smartcontractkit/chainlink-starknet/pull/434
https://github.com/smartcontractkit/chainlink-solana/pull/718
https://github.com/smartcontractkit/chainlink-cosmos/pull/405
https://github.com/smartcontractkit/chainlink-common/pull/526

### CODE REVIEW

File: core/services/chainlink/config_test.go [hunk 1]
```diff
@@ -1294,19 +1294,19 @@ func TestConfig_Validate(t *testing.T) {
...
 				- 1.TendermintURL: missing: required for all nodes
 		- 1.Nodes: missing: must have at least one node
 		- 2: 2 errors:
 			- ChainID: missing: required for all chains
 			- Nodes: missing: must have at least one node
-	- Solana: 5 errors:
+	- Solana: 4 errors:
 		- 1.ChainID: invalid value (mainnet): duplicate - must be unique
-		- 1.Nodes.1.Name: invalid value (bar): duplicate - must be unique
+	1.Nodes.1.Name: invalid value (bar): duplicate - must be unique
```
Comment (ilija42) [inline core/services/chainlink/config_test.go]: ??

File: core/services/chainlink/config_test.go [hunk 2]
```diff
@@ -1294,19 +1294,19 @@ func TestConfig_Validate(t *testing.T) {
...
 				- 0.TendermintURL: missing: required for all nodes
 				- 1.TendermintURL: missing: required for all nodes
 		- 1.Nodes: missing: must have at least one node
 		- 2: 2 errors:
 			- ChainID: missing: required for all chains
 			- Nodes: missing: must have at least one node
-	- Solana: 5 errors:
+	- Solana: 4 errors:
 		- 1.ChainID: invalid value (mainnet): duplicate - must be unique
-		- 1.Nodes.1.Name: invalid value (bar): duplicate - must be unique
```
Comment (ilija42) [inline core/services/chainlink/config_test.go]: Idk what to do with this, Solana PR that should fix this is not yet merged

Comment (jmank88) [reply core/services/chainlink/config_test.go]: - #13253

**Review by ilija42 — COMMENTED** [review]
Failing core test and Solana integration test will be fixed by an incoming PR that is being reviewed unrelated to this

**Review by archseer — APPROVED** [review]

**Review by DeividasK — APPROVED** [review]

**Pull request merged**